pkg/api/quota.go

```diff
@@ -1,20 +1,29 @@
 package api
 
 import (
+	macaron "gopkg.in/macaron.v1"
+
 	"github.com/grafana/grafana/pkg/api/response"
 	"github.com/grafana/grafana/pkg/bus"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/setting"
-	macaron "gopkg.in/macaron.v1"
 )
 
-func GetOrgQuotas(c *models.ReqContext) response.Response {
-	if !setting.Quota.Enabled {
+func (hs *HTTPServer) GetCurrentOrgQuotas(c *models.ReqContext) response.Response {
+	return hs.getOrgQuotasHelper(c, c.OrgId)
+}
+
+func (hs *HTTPServer) GetOrgQuotas(c *models.ReqContext) response.Response {
+	return hs.getOrgQuotasHelper(c, c.ParamsInt64(":orgId"))
+}
+
+func (hs *HTTPServer) getOrgQuotasHelper(c *models.ReqContext, orgID int64) response.Response {
+	if !hs.Cfg.Quota.Enabled {
 		return response.Error(404, "Quotas not enabled", nil)
 	}
-	query := models.GetOrgQuotasQuery{OrgId: c.ParamsInt64(":orgId")}
+	query := models.GetOrgQuotasQuery{OrgId: orgID}
 
-	if err := bus.DispatchCtx(c.Req.Context(), &query); err != nil {
+	if err := hs.SQLStore.GetOrgQuotas(c.Req.Context(), &query); err != nil {
```

Comment:
What's the difference between importing `sqlstore` and calling `GetOrgQuotas` directly similarly as you do in `pkg/api/org.go` and using `SQLStore` field from `HTTPServer` to call its `GetOrgQuotas`?

Reply:
I'm replacing by the method that was registered on the bus to handle the `GetOrgQuotasQuery`. `GetOrgQuotas` is a method of the `SQLStore` pointer receiver.
On the other hand `GetOrgById` and `SearchOrgs` are public methods of the sqlstore package.

In general, I think we are moving towards using methods instead of public functions relying on package variables (safer). I won't refactor `SQLStore` here :smile:

Reply:
Thanks for the context! All makes sense. And please don't refactor `SQLStore`, that wasn't my intention at all 😆 I was just curious about the difference, as the action carried out in both places seemed pretty similar (retrieve some org related info from the db).